tests/test_api.rs

```diff
@@ -513,6 +513,14 @@ fn get_isolate_from_handle() {
   check_eval(scope, Some(false), "3.3 / 3.3");
 }
 
+pub extern "C" fn backing_store_deleter_callback(
+  _data: *mut c_void,
+  _byte_length: usize,
+  _deleter_data: *mut c_void,
+) {
+  // no-op
```

Comment:
Should part of the test be to assert based on the data here? Though, assertions here would try to unwind through C++ so that'd be pretty bad. Maybe cout based assertions could be utilized?

Reply:
Not sure I understand, why should we assert the data here? This finalized callback doesn't need to be utilized since the pointer is to slice owned by Rust, it will be freed when it goes out of scope - which is pretty bad though. Maybe I should use manually allocated memory for this test using `std::alloc`, such that I can assert data before freeing it?

Reply:
Hmmm, well definitely the intended use case for this function is in general about memory more akin to `std::alloc`, ie. memory outside of Rust's borrow checker / memory management.

The current test asserts that the API can be called but not if it actually functionally works. That is, of course, more about V8 testing but one could argue that the bindings should still be tested with some code functionality as well.

So in short what I mean is that the callback could assert that the arguments are as expected: length is the same, data can be asserted to be equal to what was originally there, and that the `deleter_data` has passed through V8 unmolested (well, that is if we want to add some eg. `Box` as the data).

The eventual use-case for this API may well end up looking like this, a complete no-op as the deleter callback. But it might also be that some use-case will be found where a custom deleter callback is indeed needed. The callback argument assertions would be tests in preparation for that.